src/backend/latex/document/acmarticle.rs

```diff
@@ -0,0 +1,100 @@
+    type Image = latex::ImageGen;
+    type Label = latex::LabelGen;
+    type Pdf = latex::PdfGen;
+    type SoftBreak = latex::SoftBreakGen;
+    type HardBreak = latex::HardBreakGen;
+    type TaskListMarker = latex::TaskListMarkerGen;
+    type TableOfContents = latex::TableOfContentsGen;
+    type Bibliography = latex::BibliographyGen;
+    type ListOfTables = latex::ListOfTablesGen;
+    type ListOfFigures = latex::ListOfFiguresGen;
+    type ListOfListings = latex::ListOfListingsGen;
+    type Appendix = latex::AppendixGen;
+
+    type Paragraph = latex::ParagraphGen;
+    type Rule = latex::RuleGen;
+    type Header = latex::HeaderGen<'a>;
+    type BlockQuote = latex::BlockQuoteGen;
+    type CodeBlock = latex::CodeBlockGen;
+    type List = latex::ListGen;
+    type Enumerate = latex::EnumerateGen;
+    type Item = latex::ItemGen;
+    type FootnoteDefinition = latex::FootnoteDefinitionGen;
+    type UrlWithContent = latex::UrlWithContentGen<'a>;
+    type InterLinkWithContent = latex::InterLinkWithContentGen;
+    type HtmlBlock = latex::HtmlBlockGen;
+    type Figure = latex::FigureGen<'a>;
+
+    type TableFigure = latex::TableFigureGen<'a>;
+    type Table = latex::TableGen<'a>;
+    type TableHead = latex::TableHeadGen;
+    type TableRow = latex::TableRowGen;
+    type TableCell = latex::TableCellGen;
+
+    type InlineEmphasis = latex::InlineEmphasisGen;
+    type InlineStrong = latex::InlineStrongGen;
+    type InlineStrikethrough = latex::InlineStrikethroughGen;
+    type InlineCode = latex::InlineCodeGen;
+    type InlineMath = latex::InlineMathGen;
+
+    type Equation = latex::EquationGen<'a>;
+    type NumberedEquation = latex::NumberedEquationGen<'a>;
+    type Graphviz = latex::GraphvizGen<'a>;
+
+    fn new() -> Self {
+        AcmArticle
+    }
+
+    fn gen_preamble(&mut self, cfg: &Config, out: &mut impl Write, diagnostics: &Diagnostics<'a>) -> FatalResult<()> {
+        // TODO: itemizespacing
+        writeln!(out, "\\PassOptionsToPackage{{usenames,dvipsnames}}{{color}}")?;
+        writeln!(out, "\\PassOptionsToPackage{{pdfusetitle}}{{hyperref}}")?;
+        writeln!(out, "\\PassOptionsToPackage{{final}}{{microtype}}")?;
+        preamble::write_documentclass(cfg, out, "acmart", "nonacm,sigconf,natbib=false,pdfusetitle,")?;
+        preamble::write_packages(cfg, out)?;
+        preamble::write_fixes(cfg, out)?;
+
+        if let Some(abstract1) = &cfg.abstract1 {
+            writeln!(out, "\\begin{{abstract}}")?;
+            preamble::gen_abstract(abstract1.clone(), "abstract", &Arena::new(), AcmArticle, cfg, out, diagnostics)?;
+            writeln!(out, "\\end{{abstract}}")?;
+        }
+
+        writeln!(out)?;
+        writeln!(out, "\\begin{{document}}")?;
+        writeln!(out)?;
+
+        if cfg.title.is_some() {
+            // TODO: Warn if title isn't set but something else is
+            preamble::write_maketitle_info(cfg, out)?;
+            writeln!(out, "\\maketitle")?;
+        }
```

Comment:
`write_maketitle_info` internally only writes information that exists. This makes the check appear brittle, i.e. intentionally too lenient. I'd expect the provider of the title page content to have some method of the form  

> `has_titlepage_info(&Config) -> Result<bool>`

which can at least differentiate three cases:
* all info was provided (`Ok(true)`)
* no info was provided and no title page is to be generated (`Ok(false)`)
* partial info was provided but some necessary data was missing (`Err(_)`)

For backends that use the defines in `preamble` or known `acm` standards this could be a common method.

Reply:
I'd be fine with a method that is not completely comprehensive but this would emphasize how document classes should check and make it easier to refactor and consolidate them.

Reply:
You forgot the case that's mentioned in that todo: Info was provided that's not used by maketitle of that documentclass. Not specifying everything should be perfectly fine, I designed the templates in a way that doesn't require all titlepage properties to be set to still work (and maketitle does that as well). Providing partial info should be specifically supported by us, not everyone needs to have the `supervisor` and `advisor` fields for an article, a checklist or something simple. Those fields are only required in some cases and leaving them out shouldn't result in a diagnostic.

In the long term I'd like to have titlepage info be its own struct embedded in the `Config`. When we do that, we should be able to more easily handle all of its fields by destructuring it. That way we won't miss a place it's used when we add another field. That's also when adding those diagnostics should be easy.